sirepo/template/controls.py

```diff
@@ -5,4 +5,22 @@
 :license: http://www.apache.org/licenses/LICENSE-2.0.html
 """
 from __future__ import absolute_import, division, print_function
-from sirepo.template.madx import *
+from pykern.pkcollections import PKDict
+import sirepo.template.madx
+
+
+def background_percent_complete(report, run_dir, is_running):
+    if is_running:
+        return PKDict(
+            percentComplete=0,
+            frameCount=0,
+        )
+    return PKDict(
+        percentComplete=100,
+        frameCount=1,
+    )
+
+
+def write_parameters(data, run_dir, is_parallel):
+    data.models.externalLattice.report = ''
+    sirepo.template.madx.write_parameters(data.models.externalLattice, run_dir, is_parallel)
```

Comment:
The whole mad-x dataset is now contained in data.models.externalLattice. That way the controls app doesn't need to know about all the fields and the controls-schema.json has only the models specific to the app. The "externalLattice" could be swapped out with an elegant lattice or an epics controller without much additional work.

Reply:
Makes sense. Thanks.

Reply:
how do fixups work on this?

Reply:
In sim_data.controls, it runs the fixups on the embedded lattice:
```
    def fixup_old_data(cls, data):
        sirepo.sim_data.get_class('madx').fixup_old_data(data.models.externalLattice)
```